.github/workflows/code-analysis.yml

```diff
@@ -0,0 +1,37 @@
+name: Code Analysis
+
+on:
+  push:
+    branches: [ master, android-v2 ]
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+
+jobs:
+  sonar:
+    name: SonarScanner
+    runs-on: ubuntu-20.04
+    permissions:
+      pull-requests: read
+      actions: read
+      contents: read
+      security-events: write
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Build scanner image
+        run: |
+          make scanner/image
+      - name: Build app and scan code
```

Comment:
It would be more convenient to have separated build and scan stages.
We would know how much time each logical step takes, and if anything fails it would be easier to search for the cause.